packages/web/hooks/userIntersectionObserver.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * A hook to unable using an Intersection Observer.
+ *
+ * Wraps the logic in a try/catch because some browsers don't support IntersectionObserver
+ * and this would throw an error, rendering the app unusable for those users.
+ * * Currently only supports observing one node per usage, but could be modified to handle more.
+ *
+ * @param ref : Current code expects a div element
+ * @param rootMargin : String
+ *                   : CSS Margin syntax - sets an invisible box around the root
+ *                   : Example: '0px 0px 0px 0px'
+ * @returns Boolean : true if the DOM element crosses the threshold, otherwise false
+ */
+
+type IntersectionObserverOptions = {
+  root?: HTMLElement | null
+  rootMargin?: string
+  threshold?: number
+}
+
+const useIntersectionObserver = ({
+  root = null,
+  rootMargin = '0px',
+  threshold = 0,
+}: IntersectionObserverOptions) => {
+  const [observedElementRef, setObservedElementRef] = useState(null)
```

Comment:
Why use state rather than an actual ref here? Maybe in case the observed element changes? I guess that could happen if the element is conditionally rendered.

You might need `useState<HTMLElement>` (or `HTMLElement | null` or something) to make TSC happy though